src/login/logind-button.c

```diff
@@ -495,6 +634,9 @@ int button_open(Button *b) {
 
         (void) button_set_mask(b->name, fd);
 
+        if (!button_get_modifier_mask(b))
+                return log_warning_errno(errno, "Failed to get modifier mask on %s: %m", p);
```

Comment:
use the error code button_get_modifier_mask() returns as retval, do not use `errno` here. i.e.

```c
r = button_get_modifier_mask(b);
if (r < 0)
        return log…
``